salt/modules/cmdmod.py

```diff
@@ -761,6 +780,8 @@ def _run(
             ret["retcode"] = 0
         ret["stdout"] = out
         ret["stderr"] = err
+        if ret["stdout"] in success_stdout or ret["stderr"] in success_stderr:
+            ret["retcode"] = 0
```

Comment:
**question** Is this behavior "correct" or should it be checking retcodes? :thinking:

Reply:
It kinda look backwards to me, but I do think comparing against the output is the [correct idea](https://github.com/saltstack/salt/issues/50597). I'd expect the user to provide just a partial match, e.g.

```suggestion
        if success_stdout in ret["stdout"] or success_stderr in ret["stderr"]:
            ret["retcode"] = 0
```

Reply:
See comment above.  The string provided in either the `success_stdout` or `success_stderr` option to cmdmod is the full string or a partial string for output that should be considered a successful run.

Reply:
Thanks @garethgreenaway. Can I maybe ask for an example of the user input and output that ought to match? I feel like you are saying this should work, but it feels like the code doesn't support it:

```
>>> "foobar" in ["foo", "bar"]
False
```

Reply:
Ahh I see it now.  You're absolutely right.  This would only match if you provide the full expected string.  It would need to be something like this for the stdout:
```
if any([s for s in ret["stdout"] if any(xs in s for xs in stdout_success)]):
        ret["retcode"] = 0
```

and this for stderr:
```
if any([s for s in ret["stderr"] if any(xs in s for xs in stdout_success)]):
        ret["retcode"] = 0
```

Reply:
Wait, I thought `ret["stdout"]` and `ret["stderr"]` were strings. Are they sometimes lists? Strings are iterable, right? So if they're strings, I think we want to compare them directly, not iterate over them. Too many times I've unintentionally created lists of individual chars that way!

```
>>> [s for s in ret["stdout"]]
['f', 'o', 'o', 'b', 'a', 'r']
```

Reply:
They are. I made it more complicated than it needs to. I'm also trying to remember the reason why the call to split_input is there. Might have been to support comma separated strings, similar to the success_retcodes argument.

Reply:
Took a stab at an updated implementation that will support matching partial strings.